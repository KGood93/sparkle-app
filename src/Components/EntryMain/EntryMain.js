//

import React, {Component} from 'react'
import './EntryMain.css';
import Entry from '../Entry/Entry'
import ApiContext from '../../ApiContext'
import {findEntry} from '../../entry-helpers'

class EntryMain extends Component {
  static contextType = ApiContext
  
  render() {
    const {entry=[]} = this.context
    //console.log(entry)
    const {entryId} = this.props.match.params
    //console.log(this.props.match)
    const id = parseInt(entryId)
    const disEntry = findEntry(entry, id) || {title: '', content: '', quoteid: ''}

    //console.log(disEntry.quoteid)

    return (
        <div className='EntryPageMain'>
          <Entry
            title={disEntry.title}
            content={disEntry.content}
            quoteid={disEntry.quoteid}
          />

        </div>
    )
  }
}
export default EntryMain;