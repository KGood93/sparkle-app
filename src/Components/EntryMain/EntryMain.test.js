import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import EntryMain from './EntryMain';

describe('EntryMain Component', () => {
    it(`renders without crashing`, () => {
        const test = {
            path: "/entry/:entryId",
            url: "/entry/32",
            isExact: true,
            params: {entryId: "32"}
        };
        const wrapper = shallow(<EntryMain match={test}/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});