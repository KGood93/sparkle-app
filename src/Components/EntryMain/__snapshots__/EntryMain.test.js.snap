// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EntryMain Component renders without crashing 1`] = `
<div
  className="EntryPageMain"
>
  <Entry
    content=""
    quoteid=""
    title=""
  />
</div>
`;
